File: BackEnd/Models/Note.cs

using System;
using BackEnd.Models;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace BackEnd.Models;

public class Note
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]

    public string Id { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public NoteStatus Status { get; set; } = NoteStatus.Active;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? LastUpdatedAt { get; set;}

   
}

File: frontEnd/src/types/Note.ts
export type NoteStatus = 'Active' | 'Archived';

export interface Note {
  id: string;
  title: string;
  content: string;
  createdAt: string;
  lastUpdatedAt?: string;
  status: NoteStatus;
}

export interface NoteCreateDto {
  title: string;
  content: string;
  // Optional: assign default status in backend if not passed
  status?: NoteStatus;
}

export interface NoteUpdateDto {
  title?: string;
  content?: string;
  status: NoteStatus;
}
